### PR TITLE
Fix issue 653 (panic in MemoryDescriptor) + remove panic in UnsharedMemory

### PR DESCRIPTION
**Pull request by pventuzelo**
#Description

Based on comment from @syrusakbary [here](https://github.com/wasmerio/wasmer/pull/654#issuecomment-520908172) and previous pull request from @bjfish.
* Reformat MemoryDescriptor (new structure field, function new, function memory_type)
* Remove panic for MemoryDescriptor
* Remove panic for UnsharedMemory

# Review

- [ ] Create a short description of the the change in the CHANGELOG.md file

### CODE REVIEW

File: lib/runtime-core/src/types.rs [hunk 1]
```diff
@@ -335,16 +335,30 @@ pub struct MemoryDescriptor {
     pub maximum: Option<Pages>,
     /// This memory can be shared between wasm threads.
     pub shared: bool,
+    /// The type of the memory
```
Comment (nlewycky) [inline lib/runtime-core/src/types.rs]: Ends in a period, please.

Comment (syrusakbary) [reply lib/runtime-core/src/types.rs]:
```suggestion
    /// The type of the memory.
```

**Comment by syrusakbary** [comment]
@pventuzelo tests seem to be failing:
https://dev.azure.com/wasmerio/wasmer/_build/results?buildId=1097&view=logs&jobId=ad07b744-5a1a-5d54-7693-fcb02e666633

**Comment by pventuzelo** [comment]
my bad i forgot the spectests. 
Should work now ;)

**Comment by syrusakbary** [comment]
@pventuzelo still failing.

`make test` should work :)

**Comment by pventuzelo** [comment]
Sorry @syrusakbary, should be good now ;)
`make test` is working on my side.

**Comment by syrusakbary** [comment]
@pventuzelo checks are also failing: https://dev.azure.com/wasmerio/wasmer/_build/results?buildId=1132

The following commands should be successful:

```bash
make check
make capi
```

**Comment by pventuzelo** [comment]
ok i've run those tests as well and it's working now.

@syrusakbary  maybe we should add `check` and `capi` to the `make precommit` ?

**Pull request merged**